servers/src/main/java/tachyon/worker/TachyonWorker.java

```diff
@@ -208,6 +208,15 @@ private TachyonWorker(InetSocketAddress masterAddress, InetSocketAddress workerA
     }
     mWorkerAddress =
         new NetAddress(workerAddress.getAddress().getCanonicalHostName(), mPort, mDataPort);
+
+    // Connect to UFS before initializing the workerStorage.
+    try {
+      connectToUFS();
+    } catch (IOException ioe) {
+      LOG.error(ioe.getMessage(), ioe);
```

Comment:
Maybe change the string portion of this error to `Worker @ hostname failed to connect to the under file system`?